pkg/discovery/network/weavenet.go

```diff
@@ -34,16 +34,23 @@ func discoverWeaveNetwork(clientSet kubernetes.Interface) (*ClusterNetwork, erro
 	var clusterNetwork *ClusterNetwork
 
 	for i := range weaveNetPod.Spec.Containers {
+		var found bool
 		for _, envVar := range weaveNetPod.Spec.Containers[i].Env {
 			if envVar.Name == "IPALLOC_RANGE" {
 				clusterNetwork = &ClusterNetwork{
 					PodCIDRs:      []string{envVar.Value},
 					NetworkPlugin: cni.WeaveNet,
 				}
-
+				found = true
 				break
 			}
 		}
+		if !found {
+			clusterNetwork = &ClusterNetwork{
+				PodCIDRs:      []string{"10.32.0.0/12"},
```

Comment:
Thank you @josecastillolema for the PR. But I'm not sure if this is the approach we want to take as hard-coding the CIDRs could be error prone in future when weave-net decides to modify the default CIDR.

We have options to explicitly specify the `PodCIDR` and `ServiceCIDR` while joining the clusters. Will that option work for your use-case?

```
subctl join --clustercidr .... --servicecidr 
```

Reply:
Imho the problem is not the fact that it not detects automatically the CIDR, is super easy to provide it. The problem is that it fails to detect the CNI.
Would an approach where the **_NetworkPlugin_** was defined but the **_PodCIDRs_** set to nil be prefered? This way, the CNI would be discovered and we would not have any problems in case the default CIDR changes in the future.

Reply:
Perhaps another solution would be to retrieve the default from Weave somehow, but so far I haven’t figured out where it’s defined. I can find the unit tests that verify the default, but not the default itself!

Reply:
It seems reasonable to still report the plugin was detected as @josecastillolema suggests. If the `PodCIDRs` isn't set, subsequent code will try a generic approach to detect it. We can just define `clusterNetwork` on L34:

```Go
clusterNetwork := &ClusterNetwork{
    NetworkPlugin: cni.WeaveNet,
}
```
and then set `PodCIDRs` later.

Reply:
Agree @tpantelis .
Adjusted as requested.